frontend/src/components/PlannerChart.tsx

```diff
@@ -55,12 +55,13 @@ export const PlannerChart: React.FC<{
   if (!savedData) {
     // Add current roadmaps all tasks to the versions list that will be charted
 
+    // const customers: Customer[] | undefined   <- breaks taskUtils/calcWeightedTaskPriority if undefined
     graphTaskLists.unshift({
       name: DataKeys.OptimalRoadmap,
       tasks: [...currentRoadmap.tasks].sort(
         (a, b) =>
-          calcWeightedTaskPriority(b, customers!, currentRoadmap) -
-          calcWeightedTaskPriority(a, customers!, currentRoadmap),
+          calcWeightedTaskPriority(b, customers ?? []!, currentRoadmap) -
+          calcWeightedTaskPriority(a, customers ?? []!, currentRoadmap),
```

Comment:
Here the behaviour was consistent with the existing types, e.i. `customers` was typed as `Customer[] | undefined`
The problem was the incorrect type assertion, so I'd say the comment is not needed here

Also, the assertion is now unnecessary
```suggestion
          calcWeightedTaskPriority(b, customers ?? [], currentRoadmap) -
          calcWeightedTaskPriority(a, customers ?? [], currentRoadmap),
```